doc/source/whatsnew/v2.0.0.rst

```diff
@@ -348,6 +348,7 @@ Other API changes
 - Changed behavior of :class:`Index` constructor with sequence containing at least one ``NaT`` and everything else either ``None`` or ``NaN`` to infer ``datetime64[ns]`` dtype instead of ``object``, matching :class:`Series` behavior (:issue:`49340`)
 - :func:`read_stata` with parameter ``index_col`` set to ``None`` (the default) will now set the index on the returned :class:`DataFrame` to a :class:`RangeIndex` instead of a :class:`Int64Index` (:issue:`49745`)
 - Changed behavior of :class:`Index` constructor with an object-dtype ``numpy.ndarray`` containing all-``bool`` values or all-complex values, this will now retain object dtype, consistent with the :class:`Series` behavior (:issue:`49594`)
+- Changed behavior of :class:`Series` and :class:`DataFrame` constructors when given an integer dtype and floating-point data that is not round numbers, this now raises ``ValueError`` instead of silently retaining the float dtype; do ``Series(data).astype(dtype)`` or ``DataFrame(data).astype(dtype)`` to get the old behavior (:issue:`49599`)
```

Comment:
This note confuses me a bit. The old Series behavior just returns ``float64``?

But if you do ````Series(data).astype(dtype)`` then you get ``dtype`` (for example ``int64``), which cuts the decimals and returns integers?

Reply:
correct.  in main:

```
arr = np.random.randn(2)

>>> pd.Series(arr, dtype=np.int64).dtype
dtype('float64')

>>> pd.Series(arr).astype(np.int64).dtype
dtype('int64')
```

Reply:
Ok, but the old behaviour then would be to do: ``pd.Series(arr, dtype=np.float64)``

Nothing changed when you want to actually get ``int64``? You have to call astype now and also after this pr?

Reply:
> Nothing changed when you want to actually get int64? You have to call astype now and also after this pr?

correct

Reply:
Shouldn't we rephrase the last part then? For example:

> ; do ``DataFrame(data).astype(dtype)`` to get the old behavior (:issue:`49599`)


Not sure if I am overly picky though...

Reply:
sure. what do you have in mind?

Reply:
I'd just remove "``Series(data).astype(dtype)`` or" and only mention the DataFrame.astype part

Reply:
> - Changed behavior of :class:`Series` and :class:`DataFrame` constructors when given an integer dtype and floating-point data that is not round numbers, this now raises ``ValueError`` instead of silently retaining the float dtype; do ``DataFrame(data).astype(dtype)`` to get the old behavior (:issue:`49599`)

Reply:
that seems weird to me since we're explicitly talking about both at the beginning of the line

Reply:
oh i see.  will update